bundles/org.openhab.binding.pentair/src/main/java/org/openhab/binding/pentair/internal/PentairControllerSchedule.java

```diff
@@ -0,0 +1,296 @@
+        }
+
+        end = min;
+        dirty = true;
+
+        return true;
+    }
+
+    public boolean setScheduleType(ScheduleType type) {
+        if (this.type == type) {
+            return true;
+        }
+
+        this.type = type;
+        dirty = true;
+
+        return true;
+    }
+
+    public boolean setScheduleType(String typestring) {
+        ScheduleType scheduleType;
+
+        try {
+            scheduleType = ScheduleType.valueOf(typestring);
+        } catch (IllegalArgumentException e) {
+            return false;
+        }
+
+        return setScheduleType(scheduleType);
+    }
+
+    public boolean setDays(String d) {
+        final String dow = "SMTWRFY";
+
+        days = 0;
+        for (int i = 0; i <= 6; i++) {
+            if (d.indexOf(dow.charAt(i)) >= 0) {
+                days |= 1 << i;
+            }
+        }
+
+        dirty = true;
+
+        return true;
+    }
+
+    public @Nullable PentairPacket getWritePacket(int controllerid, int preamble) {
+        byte[] packet = { (byte) 0xA5, (byte) preamble, (byte) controllerid, (byte) 0x00 /* source */, (byte) 0x91,
+                (byte) 7, (byte) id, (byte) circuit, (byte) (start / 60), (byte) (start % 60), (byte) (end / 60),
+                (byte) (end % 60), (byte) days };
+        PentairPacket p = new PentairPacket(packet);
+
+        switch (type) {
+            case NONE:
+                p.setByte(STARTH, (byte) 0);
+                p.setByte(STARTM, (byte) 0);
+                p.setByte(ENDH, (byte) 0);
+                p.setByte(ENDM, (byte) 0);
+                p.setByte(CIRCUIT, (byte) 0);
+                p.setByte(DAYS, (byte) 0);
+                break;
+
+            case NORMAL:
+                break;
+
+            case ONCEONLY:
+                p.setByte(ENDH, (byte) 26);
+                p.setByte(ENDM, (byte) 0);
+                break;
+            case EGGTIMER:
+                p.setByte(STARTH, (byte) 25);
+                p.setByte(STARTM, (byte) 0);
+                p.setByte(DAYS, (byte) 0);
+                break;
+            case UNKNOWN:
+                return null;
+        }
+
+        return p;
+    }
+
+    public String getDays() {
+        final String dow = "SMTWRFY";
+        String str = "";
+
+        for (int i = 0; i <= 6; i++) {
+            if ((((days >> i) & 0x01)) == 0x01) {
+                str += dow.charAt(i);
+            }
+        }
+
+        return str;
+    }
+
+    @Override
+    public String toString() {
+        String str = String.format("%s,%d,%02d:%02d,%02d:%02d,%s", getScheduleTypeStr(), circuit, start / 60,
+                start % 60, end / 60, end % 60, getDays());
+
+        return str;
+    }
+
+    public boolean fromString(String str) {
+        final String rePattern = "^(NONE|NORMAL|EGGTIMER|ONCEONLY),(\\\\d+),(\\\\d+):(\\\\d+),(\\\\d+):(\\\\d+),([SMTWRFY]+)";
+
+        String schedulestr = str.toUpperCase();
+
+        final Pattern ptn = Pattern.compile(rePattern);
```

Comment:
Please make this pattern a static final field.

Reply:
that gives me an "illegal modifier for Pattern ptn; only final is permitted"

Reply:
You have to make it a static final **field** instead of a local variable.